src/app/components/common/final-object.interface.ts

```diff
@@ -16,6 +16,7 @@ export interface ImageElement {
   duration: number;              // duration of film as number
   fileName: string;              // for opening the file
   fileSize: number;              // file size in bytes
+  tags?: string[];               // tags associated with this particular file
```

Comment:
I'm eager to have other tags like `stars` and `timeAdded` and `timesPlayed` etc.
I'd prefer those be separate fields, rather than get mixed with `tags`.

What did you have in mind? 😃

Reply:
I was just thinking of future proofing in case we ever wanted to add more metadata to individual tags (priority/order maybe)! 👍 But it wouldn't be too hard of a change if we ever needed to!

Reply:
I'm planning to add a panel which will show all the currently-added tags (next PR 😉 ) and there will be an option to sort by frequency or alphabetical. 

Now that you mention it, it may be worthwhile to also allow the tags along each video to also be sorted by frequency (not just alphabetical order - as my previous intention).

I see there's room for allowing different color for tags (etc) in the future, but I'd rather not go there (meta stuff attached to the tags) in this release or immediate future. 

As you say - eventually - if needed - we can upgrade the field.